proposals/target-typed-new.md

```diff
@@ -0,0 +1,100 @@
+
+# Target-typed `new` expressions
+
+* [x] Proposed
+* [x] [Prototype](https://github.com/alrz/roslyn/tree/features/implicit-new)
+* [ ] Implementation
+* [ ] Specification
+
+## Summary
+[summary]: #summary
+
+Do not require type specification for constructors when the type is known. 
+
+## Motivation
+[motivation]: #motivation
+
+Allow field initialization without duplicating the type.
+```cs
+Dictionary<string, List<int>> field = new() {
+    { "item1", new() { 1, 2, 3 } }
+};
+```
+Allow omitting the type when it can be inferred from usage.
+```cs
+XmlReader.Create(reader, new() { IgnoreWhitespace = true });
+```
+
+## Detailed design
+[design]: #detailed-design
+
+The *object_creation_expression* syntax would be modified to make the *type* optional when parentheses are present. This is required to address the ambiguity with *anonymous_object_creation_expression*.
+```antlr
+object_creation_expression
+    : 'new' type? '(' argument_list? ')' object_or_collection_initializer?
+    | 'new' type object_or_collection_initializer
+    ;
+```
+The target-type is resolved through a conversion that only exists if and only if there exists a single best accessible constructor with regard to provided arguments. Target of the initialization is determined after the conversion succeeded.
+
+The type of the expression would be inferred from the target-type which would be required to be one of the following:
+
+- **Any struct type**
+- **Any reference type**
+- **Any type parameter** with a constructor constraint
+
+The following types do not contribute to the conversion:
+
+- **Enum types:** not all enum types contain the constant zero, so it should be desirable to use the explicit enum member.
+- **Delegate types:** if the type is inferrable, an anonymous function can already be used.
+- **Interface types:** this is a niche feature and it should be preferable to explicitly mention the type.
+- **Tuple types:** tuple literals should be used for such expressions.
+- **Array types:** arrays need a special syntax to provide the length.
+
+
+All the other types that are not permitted in the *object_creation_expression* are excluded as well, for instance, pointer types.
+
+>**Open Issue:** What other types should be excluded?
+
+
+
+Note that any restriction on permitted types would raise the success rate of the overload resolution. For example, if we exclude primitive types, the following would successfully compile.
```

Comment:
Rather than excluding primitive types, I suggest we ought to exclude the default constructor for a value type (unless the constructor appears explicitly in metadata). That would have the effect of excluding the primitive types and the default constructor or most value types as well. If you wanted to use the default value of such types you could write `default`.